uktrade/data-hub-frontend
# Refactor contact form to react

## Description of change

Refactors the add/edit contact page from Nunjucks to React. Along the way, I have:

- Tweaked the `Form/elements/FieldAddress` component so that it conditionallyrenders a dropdown for its country field
- Fixed an issue in the `ReferrerLink` component, which was breaking some tests
- Added the `superHeading` option to the `LocalHeader` component
- Introduced the `State` component
- Introduced the `Resource` component together with its `Resource/Company`, `Resource/Contact` and `Resource/Countries` derivations two of which are used in the `ContactForm`
- Introduced the `Form/elements/FieldCountrySelect` which uses `Resource/Company`
- Introduced the `email` form validator in `src/client/components/Form/validators.js`
- Added the `idProp` parameter to `multiInstance` so that the `Resource` component could be implemented with it
- Added the `checkRadioGroup` and `getSubmitButtonByLabel` Cypress commands (we are not using Cypress Testing Library)

I realize that this PR is way beyond the 400 LOC limit, but the only other way would be way to introduce the new components in separate PRs without actually being used anywhere, which would raise a lot of questions and require lenghty explanation. But should this be really hard to review, it can still be broken up into smaller PRs.

## Test instructions

Nothing should change by this pr. The add/edit contact form should work as before.
